### ISSUE TITLE
Regarding the errors encountered when exporting a single channel model to TensorRT

### ISSUE THREAD

**Issue by tongchangD** [ultralytics/ultralytics#16556]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Bug

my single Model structure
```
                   from  n    params  module                                       arguments                     
  0                  -1  1       176  ultralytics.nn.modules.conv.Conv             [1, 16, 3, 2]                 
  1                  -1  1      4672  ultralytics.nn.modules.conv.Conv             [16, 32, 3, 2]                
  2                  -1  1      7360  ultralytics.nn.modules.block.C2f             [32, 32, 1, True]             
  3                  -1  1     18560  ultralytics.nn.modules.conv.Conv             [32, 64, 3, 2]                
  4                  -1  2     49664  ultralytics.nn.modules.block.C2f             [64, 64, 2, True]             
  5                  -1  1     73984  ultralytics.nn.modules.conv.Conv             [64, 128, 3, 2]               
  6                  -1  2    197632  ultralytics.nn.modules.block.C2f             [128, 128, 2, True]           
  7                  -1  1    295424  ultralytics.nn.modules.conv.Conv             [128, 256, 3, 2]              
  8                  -1  1    460288  ultralytics.nn.modules.block.C2f             [256, 256, 1, True]           
  9                  -1  1    164608  ultralytics.nn.modules.block.SPPF            [256, 256, 5]  
 10                  -1  1         0  torch.nn.modules.upsampling.Upsample         [None, 2, 'nearest']          
 11             [-1, 6]  1         0  ultralytics.nn.modules.conv.Concat           [1]                           
 12                  -1  1    148224  ultralytics.nn.modules.block.C2f             [384, 128, 1]                 
 13                  -1  1         0  torch.nn.modules.upsampling.Upsample         [None, 2, 'nearest']          
 14             [-1, 4]  1         0  ultralytics.nn.modules.conv.Concat           [1]                           
 15                  -1  1     37248  ultralytics.nn.modules.block.C2f             [192, 64, 1]                  
 16                  -1  1         0  torch.nn.modules.upsampling.Upsample         [None, 2, 'nearest']          
 17             [-1, 2]  1         0  ultralytics.nn.modules.conv.Concat           [1]                           
 18                  -1  1      9408  ultralytics.nn.modules.block.C2f             [96, 32, 1]                   
 19                  -1  1      9280  ultralytics.nn.modules.conv.Conv             [32, 32, 3, 2]                
 20            [-1, 15]  1         0  ultralytics.nn.modules.conv.Concat           [1]                           
 21                  -1  1     31104  ultralytics.nn.modules.block.C2f             [96, 64, 1]                   
 22                  -1  1     36992  ultralytics.nn.modules.conv.Conv             [64, 64, 3, 2]                
 23            [-1, 12]  1         0  ultralytics.nn.modules.conv.Concat           [1]                           
 24                  -1  1    123648  ultralytics.nn.modules.block.C2f             [192, 128, 1]                 
 25                  -1  1    147712  ultralytics.nn.modules.conv.Conv             [128, 128, 3, 2]              
 26             [-1, 9]  1         0  ultralytics.nn.modules.conv.Concat           [1]                           
 27                  -1  1    493056  ultralytics.nn.modules.block.C2f             [384, 256, 1]                 
 28    [18, 21, 24, 27]  1    509092  ultralytics.nn.modules.head.Segment          [5, 32, 64, [32, 64, 128, 256]]
```

I have successfully exported a three-channel int8 trt model,

but in single-channel, it report the following error

```
[TRT] [E] 10: [optimizer.cpp::computeCosts::3728] Error Code 10: Internal Error (Could not find any implementation for node Conv_233 + PWN(PWN(Sigmoid_234), Mul_235).)
```

### Environment

```
Ultralytics YOLOv8.2.86 🚀 Python-3.8.19 torch-1.12.0 CUDA:0 (NVIDIA GeForce RTX 3060, 12042MiB)

onnx                     1.15.0
onnxslim                 0.1.34
tensorrt                 8.5.3.1
```

### Minimal Reproducible Example

```
yolo export model=./20203.pt format=engine int8=True  data=./data.yaml
```

### Are you willing to submit a PR?

- [X] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @tongchangD, thank you for reaching out about the issue you're experiencing with exporting a single-channel model to TensorRT 🚀. This is an automated response to help guide you while an Ultralytics engineer also reviews your issue.

For 🐛 Bug Reports, please ensure you have included a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/). This helps us better diagnose and resolve the issue.

### Quick Tips:

1. **Documentation**: Have you checked the [Docs](https://docs.ultralytics.com) for relevant guides? You might find solutions in sections such as [Exporting models](https://docs.ultralytics.com/modes/export/).

2. **Upgrade**: Make sure your setup is fully up-to-date. Upgrade using the command below to ensure you're running the latest version:

   ```bash
   pip install -U ultralytics
   ```

3. **Environment**: Could you verify if the issue persists across different environments? You can try the following platforms with pre-installed dependencies:

   - **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/ultralytics/yolov8"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>

4. **Community**: Join our community for more interactive help:
   - [Discord](https://ultralytics.com/discord) for real-time support 🎧
   - [Discourse](https://community.ultralytics.com) for in-depth discussions
   - [Subreddit](https://reddit.com/r/ultralytics) for knowledge sharing

### Status:

Check our [CI status](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule) to see if this issue may have already been addressed in the latest builds.

Meanwhile, an engineer will look into your specific issue soon. Stay tuned! 😊

**Comment by tongchangD** [comment]
[weights.zip](https://github.com/user-attachments/files/17187381/weights.zip)

single channel model is that

**Comment by glenn-jocher** [comment]
@tongchangD it seems you're trying to export a single-channel model to TensorRT. Ensure your model's input dimensions are correctly set for single-channel data. If the issue persists, consider checking TensorRT compatibility with your specific model configuration.

**Comment by tongchangD** [comment]
@glenn-jocher  ok，thanks

**Comment by glenn-jocher** [comment]
You're welcome! If you have any more questions, feel free to ask.
